docs/architecture_overview.md

```diff
@@ -0,0 +1,130 @@
+
+### Assignment
+An `Assignment` can be thought of the minimal subgoal of a given `Task`. If your `Task` is to *Annotate [these] images*, one such subgoal could be to *Annotate [this] image*. It may be appropriate to batch these, such that the assignment is actually to *Annotate [these 5] images*, but the goal here is to ensure that an `Assignment` represents a reasonable amount of work to accomplish, but small enough to be reviewed as a whole.
+
+In some cases, an `Assignment` may relate to something that needs to be done by multiple workers at once, such as a cooperative task. In other cases, an `Assignment` may relate to wanting multiple different workers to attempt the same task in order to get consensus. We refer to this as a task being *concurrent* or not. In either case, however, Mephisto will prevent any given `Worker` from participating in any `Assignment` more than once.
+### Unit
+The `Unit` is the smallest bit of work that is part of a task, in that it comprises a single worker's contribution to that task. It has the required information for initializing a specific instance of work to the worker, and is responsible for having up-to-date information about the real-life status of that work.
+
+Some `Assignment`'s may have just one `Unit`, should the task be something that can be completable by just one worker and only needs completing once. Some `Assignment`'s, like a conversation, may be comprised of two (or more) `Unit`'s, each representing one worker's place in the conversation.
+## Workers
+In this section we discuss workers and their qualifications. The `Worker` and `Agent` classes specifically refer to a long-term worker and a worker's specific work on a single `Unit` respectively. The `Qualification` class refers to a kind of qualification that a `Worker` can get, and they are received in the form of a `GrantedQualification`. The goal for Mephisto's interaction with workers is to keep long-term statistics for individual workers, keep data for specific `Unit`'s organized, and ensure worker eligibility for tasks where possible.
+
+### Worker
+The Mephisto `Worker` object keeps track of a specific person after they've interacted with Mephisto. The `Worker` class is able to track overall pay, acceptance rates, and other trends based on the data that is linked to it. A `Worker` can also be granted qualifications. Mephisto uses the `Worker` class primarily to determine eligibility for specific `Unit`'s, and it exposes other functions as helpers for analysis.
+### Agent
+The `Agent` object can be thought of as an instance of a `Worker` paired to a specific `Unit`, in that it tracks the work of the human on the other end as they complete the `Unit`. It is responsible for tracking the state of the work done, as well as saving the data associated with the work. Over time, a `Worker` that continues to do tasks will end up having many `Agent`'s.
```

Comment:
Is it safe to say that a Worker will have as many Agents as Units they have worked on?